packages/protocol/contracts/common/FixidityLib.sol

```diff
@@ -0,0 +1,335 @@
+pragma solidity ^0.5.0;
+
+
+/**
+ * @title FixidityLib
+ * @author Gadi Guy, Alberto Cuesta Canada
+ * @notice This library provides fixed point arithmetic with protection against
+ * overflow.
+ * All operations are done with int256 and the operands must have been created
+ * with any of the newFrom* functions, which shift the comma digits() to the
+ * right and check for limits.
+ * When using this library be sure of using maxNewFixed() as the upper limit for
+ * creation of fixed point numbers. Use maxFixedMul(), maxFixedDiv() and
+ * maxFixedAdd() if you want to be certain that those operations don't
+ * overflow.
+ */
+library FixidityLib {
+
+  struct Fraction {
+    uint256 value;
+  }
+
+  /**
+   * @notice Number of positions that the comma is shifted to the right.
+   */
+  function digits() internal pure returns(uint8) {
+    return 24;
+  }
+
+  uint256 private constant FIXED1_INT = 1000000000000000000000000;
+
+  /**
+   * @notice This is 1 in the fixed point units used in this library.
+   * @dev Test fixed1() equals 10^digits()
+   * Hardcoded to 24 digits.
+   */
+  function fixed1() internal pure returns(Fraction memory) {
+    return Fraction(FIXED1_INT);
+  }
+
+  function wrap(uint256 x) internal pure returns (Fraction memory) {
+    return Fraction(x);
+  }
+
+  function unwrap(Fraction memory x) internal pure returns (uint256) {
+    return x.value;
+  }
+
+  /**
+   * @notice The amount of decimals lost on each multiplication operand.
+   * @dev Test mulPrecision() equals sqrt(fixed1)
+   * Hardcoded to 24 digits.
+   */
+  function mulPrecision() internal pure returns(uint256) {
+    return 1000000000000;
+  }
+
+  /**
+   * @notice Maximum value that can be represented in an int256
+   * @dev Test maxUint256() equals 2^255 -1
+   */
+  function maxUint256() internal pure returns(uint256) {
+    return 115792089237316195423570985008687907853269984665640564039457584007913129639935;
+  }
+
+  /**
+   * @notice Maximum value that can be converted to fixed point. Optimize for
+   * @dev deployment.
+   * Test maxNewFixed() equals maxUint256() / fixed1()
+   * Hardcoded to 24 digits.
+   */
+  function maxNewFixed() internal pure returns(uint256) {
+    return 57896044618658097711785492504343953926634992332820282;
+  }
+
+  /**
+   * @notice Maximum value that can be safely used as an addition operator.
+   * @dev Test maxFixedAdd() equals maxUint256()-1 / 2
```

Comment:
Are all of the cases listed here and in the other natspecs tested?

Reply:
Nope, going to go through these, see which of them it makes sense to test and write tests.